corehq/apps/app_manager/views/modules.py

```diff
@@ -1155,7 +1155,7 @@ def edit_module_detail_screens(request, domain, app_id, module_unique_id):
         try:
             detail = getattr(module, '{0}_details'.format(detail_type))
         except AttributeError:
-            return HttpResponseBadRequest("Unknown detail type '%s'" % detail_type)
+            return HttpResponseBadRequest("Unknown detail type '%s'" % strip_tags(detail_type))
```

Comment:
I think this is good. A possible alternative would be to escape the user input correctly so that when rendered as html it is displayed back to them exactly how they put it in. To do that you'd do something like

```python
format_html("Unknown detail type '{}'", detail_type)
```

In this case since it's just an error message, there's not a lot riding on it either way, so the way you chose to do it seems totally good to me.

Reply:
Thanks for the review @dannyroberts 

I did use format_html first, as you suggested over email, though I did it incorrectly, which made it ineffective  and hence looked for the alternate strip_tags approach.
`("Unknown detail type '{}'", format_html(detail_type))`

Updated now to use format_html correctly.